### PR TITLE
feat: upgrade to Angular 14

### PR DESCRIPTION
**Pull request by WeizhengSap**
Main changes:

    bump @angular/ packages from v13.3 to v14.2
    bump typescript from v4.6 to v4.7
    bump ng-packagr from v13.3 to v14
    bump relative packages (e.g. @nguniversal, @ngrx, ng-select etc.) to the highest version compatible with ng14
    upgrade some other 3rd party dependencies

### CODE REVIEW

File: scripts/install/config.default.sh [hunk 1]
```diff
@@ -51,7 +51,7 @@ CLONE_DIR="clone"
 INSTALLATION_DIR="apps"
 E2E_TEST_DIR=${CLONE_DIR}/projects/storefrontapp-e2e-cypress
 
-ANGULAR_CLI_VERSION='^13.3.0'
+ANGULAR_CLI_VERSION='^14.1.3'
```
Comment (plabadie) [inline scripts/install/config.default.sh]: I think we missed this one for the 14.2 upgrade.

File: feature-libs/checkout/base/occ/public_api.ts [hunk 2]
```diff
@@ -4,4 +4,6 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-export * from './index';
+export * from './adapters/index';
+export * from './checkout-occ.module';
+export * from './model/index';
```
Comment (plabadie) [inline feature-libs/checkout/base/occ/public_api.ts:9]: delete the corresponding index.ts

File: feature-libs/organization/order-approval/public_api.ts [hunk 3]
```diff
@@ -4,4 +4,13 @@
...
+export * from './components/index';
+export * from './core/connectors/index';
+export * from './core/guards/index';
+export * from './core/model/index';
+export * from './core/services/index';
+export * from './core/store/actions/index';
+export * from './core/store/order-approval-state';
+export * from './core/store/selectors/index';
+export * from './occ/index';
+export * from './order-approval.module';
```
Comment (plabadie) [inline feature-libs/organization/order-approval/public_api.ts:16]: Likely need to delete the associated index.ts file

**Review by plabadie — CHANGES_REQUESTED** [review]
See the 2 small comments about index.ts files.

**Review by plabadie — APPROVED** [review]

File: scripts/install/functions.sh [hunk 4]
```diff
@@ -80,48 +80,49 @@ function update_projects_versions {
...
-        ng add --skip-confirmation @spartacus/product-configurator@${SPARTACUS_VERSION} --interactive false --features="CPQ-Configurator"
+        ng add @spartacus/product-configurator --skip-confirmation --no-interactive --features "CPQ-Configurator"
     fi
 }
 
 # Don't install b2b features here (use add_b2b function for that)
 function add_feature_libs {
-  ng add --skip-confirmation @spartacus/tracking@${SPARTACUS_VERSION} --interactive false --features="TMS-GTM" --features="TMS-AEPL"
-  ng add --skip-confirmation @spartacus/qualtrics@${SPARTACUS_VERSION} --interactive false
+  ng add @spartacus/tracking --skip-confirmation --no-interactive --features "TMS-GTM" --features "TMS-AEPL"
```
Comment (RadhepS) [inline scripts/install/functions.sh:116]: why was @{spartacus_version} removed?

**Pull request merged**